Add CONTRIBUTING.md with blog post creation instructions (#44)
spyder-ide/spyder-website: README.md

# Spyder Website

> ![Spyder Logo](static/assets/media/spyder-logomark.png)

Reviewer: We should use the [Readme banner here](https://github.com/spyder-ide/spyder-docs/blob/master/doc/_static/images/spyder_readme_banner.png) that is much better proportioned and explicitly designed as a banner, and IMO as a banner it would look better above the main title.

[![Deployed on GitHub Pages](https://img.shields.io/badge/Deployed%20on-GitHub%20Pages-blue.svg)](https://pages.github.com/)

🌐 **Live Site**: [www.spyder-ide.org](https://www.spyder-ide.org)

## About

This repository contains the source code for the official Spyder IDE website. The site serves as:

- **Information hub** for Spyder features and capabilities
- **Download center** for installers and releases
- **Blog platform** for project updates and community news
- **Documentation gateway** linking to guides and resources
- **Community showcase** highlighting contributors and achievements

## Tech Stack

This website is built with modern web technologies:

- **Framework**: [SvelteKit](https://kit.svelte.dev/) - Full-stack web framework
- **Styling**: [TailwindCSS](https://tailwindcss.com/) - Utility-first CSS framework
- **Content**: [mdsvex](https://mdsvex.pngwn.io/) - Markdown processor for Svelte
- **Icons**: [Svelte Icons Pack](https://leshak.github.io/svelte-icons-pack/) - Icon library
- **Deployment**: [GitHub Pages](https://pages.github.com/) - Static site hosting

> - **CI/CD**: GitHub Actions - Automated deployment

Reviewer: ```suggestion
- **CI/CD**: [GitHub Actions](https://github.com/features/actions) - Automated deployment
```

Not sure why ChatGPT didn't link to this one like the rest :)

## Features

- ✨ **Modern Design** - Clean, responsive interface
- 📱 **Mobile Optimized** - Works perfectly on all devices
- 🚀 **Fast Performance** - Static site generation for speed
- 📝 **Blog System** - Dynamic blog with author management
- 🎨 **Component Library** - Reusable UI components
- 🌍 **Internationalization** - Multi-language support ready
- ♿ **Accessibility** - WCAG compliant design

> - 🔍 **SEO Optimized** - Meta tags and structured data

Reviewer: Sounds more like an advertising brochure rather than a readme... 🤔

May as well add [![Blazing Fast](https://badgen.now.sh/badge/speed/blazing%20%F0%9F%94%A5/green)](https://news.ycombinator.com/item?id=23874245) while you're at it 😅

## Quick Start

### Prerequisites

- [Node.js](https://nodejs.org/) (version 18 or higher)

> - [npm](https://www.npmjs.com/) (comes with Node.js)

Reviewer: This should be in the contributing guide (as this should be a subset of content there)

> ### Development Setup

Reviewer: ```suggestion
### Build locally
```

Since that's really just what this does, not actually set the user up for development (forking, branching, etc.)

1. **Clone the repository**:

>    ```bash

Reviewer: ```suggestion
   ```shell
```

Since these are generic shell commands rather than bash specific

   git clone https://github.com/spyder-ide/spyder-website.git
   cd spyder-website
   ```

2. **Install dependencies**:

>    ```bash

Reviewer: ```suggestion
   ```shell
```

   npm install
   ```

3. **Start development server**:

>    ```bash

Reviewer: ```suggestion
   ```shell
```

   npm run dev
   ```
   
   The site will be available at [http://localhost:5173](http://localhost:5173)

### Available Scripts

| Command                           | Description                              |
| --------------------------------- | ---------------------------------------- |
| `npm run dev`                     | Start development server with hot reload |
| `npm run build`                   | Build the site for production            |
| `npm run preview`                 | Preview the production build locally     |
| `npm run generate-og-images`      | Generate Open Graph images               |
| `npm run generate-contrib-data`   | Generate contributor data                |
| `npm run generate-hubspot-data`   | Generate HubSpot integration data        |
| `npm run generate-funding-schema` | Generate funding schema                  |

## Project Structure

> ```

Reviewer: ~~~suggestion
```text
~~~

To be explict

spyder-website/
├── src/

> │   ├── routes/                 # Page routes and content

Reviewer: ```suggestion
│   ├── routes/                # Page routes and content
```

Indent off by one

│   │   ├── config/            # Configuration and data
│   │   ├── i18n/              # Internationalization
│   │   ├── store/             # State management
│   │   └── utils/             # Utility functions
│   ├── app.css                # Global styles
│   └── app.html               # HTML template
├── static/                    # Static assets
│   └── assets/                # Images, fonts, media
├── scripts/                   # Build and utility scripts
└── ci/                        # CI/CD scripts
```

## Deployment

This website is automatically deployed to GitHub Pages using GitHub Actions. The deployment process:

1. **Triggers** on push to `main` branch
2. **Builds** the static site using SvelteKit
3. **Generates** dynamic content (OG images, contributor data, etc.)
4. **Deploys** to GitHub Pages

### Manual Deployment

For manual deployment or testing:

```bash
# Build the site
npm run build

# Preview the built site
npm run preview
```

The built site will be available at [http://localhost:5174](http://localhost:5174)

## Contributing

We welcome contributions from the community! Please see our [Contributing Guide](CONTRIBUTING.md) for details on:

> - Setting up the development environment
> - Creating blog posts
> - Making code contributions
> - Submitting pull requests
> - Reporting issues

Reviewer: ```suggestion
- Reporting issues
- Setting up the development environment
- Creating blog posts
- Making code contributions
- Submitting pull requests
```

Re-order to match the integrated version

### Quick Contribution Steps

> 1. Fork this repository

Reviewer: ```suggestion
1. Fork and clone this repository
```

ChatGPT forgot something here...

> 2. Create a feature branch (`git checkout -b feature/amazing-feature`)

Reviewer: ```suggestion
2. Create a topic branch (`git switch -c topic-branch-name-here`)
```

- Like many I've moved to calling these topic branches, since they may contain docs, blog posts, infra changes, bug fixes, improvements, etc. not just features
- Don't specify a particular format for branch names that we don't actually use
- Modernize Git syntax

3. Make your changes
4. Test locally (`npm run dev`)
5. Commit your changes (`git commit -m 'Add amazing feature'`)

> 6. Push to your branch (`git push origin feature/amazing-feature`)

Reviewer: ```suggestion
6. Push your branch to your fork (`git push -u origin topic-branch-name-here`)
```

- Correct the description of what's actually going on here
- Forgot `-u` (`--set-upstream`)
- Make the topic branch change from above

> 7. Open a Pull Request

Reviewer: ```suggestion
7. Open a [Pull Request](https://github.com/spyder-ide/spyder-website/pull/new)
```

## Community

- 🐍 **Main Project**: [Spyder IDE](https://github.com/spyder-ide/spyder)

> - 💬 **Discussions**: [GitHub Discussions](https://github.com/spyder-ide/spyder/discussions)

Reviewer: Uh, no, we don't use this and it 404s.

> - 🐛 **Issues**: [Bug Reports](https://github.com/spyder-ide/spyder/issues)

Reviewer: ```suggestion
- 🐛 **Issues**: [Bug Reports](https://github.com/spyder-ide/spyder-website/issues)
```

This should presumably be issues for the _website_, no?

- 📧 **Mailing List**: [Google Groups](https://groups.google.com/group/spyderlib)
- 🐦 **Twitter**: [@Spyder_IDE](https://twitter.com/Spyder_IDE)

## License

> This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details.

Reviewer: ```suggestion
This project is released under the MIT License - see the [LICENSE.md](./LICENSE.md) file for details.
```

- Avoid highly proximate reptition

## Acknowledgments

- Built with ❤️ by [@conradolandia](https://github.com/conradolandia) and the Spyder development team
- Special thanks to all [contributors](https://github.com/spyder-ide/spyder-website/graphs/contributors)
- Powered by [SvelteKit](https://kit.svelte.dev/) and [TailwindCSS](https://tailwindcss.com/)
- Hosted on [GitHub Pages](https://pages.github.com/)

> ---
> 
> *Made with love for the Python community*

Reviewer: Seems to duplicate the "made with ❤️ " just above?

```suggestion
```
